### ISSUE TITLE
Specifying development version of snscrape in requirements.txt

### ISSUE THREAD

**Issue by hiro-551010** [JustAnotherArchivist/snscrape#273]
I installed it with it.
```
pip3 install snscrape
pip3 install git+https://github.com/JustAnotherArchivist/snscrape.git
```
And I did this.
```
pip freeze > requirements.txt
```
It was saved like this.
```
snscrape==0.3.5.dev123+gc76f163
```
I deploy to heroku, but I get an error.
```
remote: ERROR: Could not find a version that satisfies the requirement snscrape==0.3.5.dev123+gc76f163 (from -r /tmp/build_0b4e46aa/requirements.txt (line 19)) (from versions: 0.1.3, 0.2.0, 0.3.0, 0.3.1, 0.3.2, 0.3.3, 0.3.4)
remote: ERROR: No matching distribution found for snscrape==0.3.5.dev123+gc76f163 (from -r /tmp/build_0b4e46aa/requirements.txt (line 19))
```
What can I do to avoid this?

Thank you in advance.

**Comment by JustAnotherArchivist** [comment]
Presumably that attempts to fetch from PyPI, which obviously doesn't have the dev version. You can put this in your `requirements.txt` to install it from Git instead:

    git+https://github.com/JustAnotherArchivist/snscrape.git#egg=snscrape

If you want to install a specific version of it, you can add `@ref` after `snscrape.git` where `ref` is a reference to a commit in the repository, e.g. a branch name or a commit ID.

So, to force it to that specific commit:

    git+https://github.com/JustAnotherArchivist/snscrape.git@c76f163#egg=snscrape

(Note that you probably won't be able to run snscrape on Heroku though: #79)

**Comment by hiro-551010** [comment]
Thank you for changing the title to something more appropriate!
i try it.Bad, it didn't work.
```
git+https://github.com/JustAnotherArchivist/snscrape.git@c76f163#egg=snscrape
```
So,I've edited the requirements.txt like this.
```
snscrape @ git+https://github.com/JustAnotherArchivist/snscrape.git
```
It's up and running!

**Comment by JustAnotherArchivist** [comment]
Sounds like Heroku or Django is doing something weird then. The notation I mentioned definitely works in a standard file with `pip install -r requirements.txt` (although the full commit hash would be preferred to the truncated version for efficiency reasons per the documentation). Just verified that with pip 21.2.4.

The syntax you mention appears to be newer, added in pip 19.1, and is called a 'direct reference' (PEP 440). Note that you should include the commit ID if you want to make sure the installation is reproducible.

So in summary, the two valid options are:

Direct reference (since pip 19.1):

    snscrape @ git+https://github.com/JustAnotherArchivist/snscrape.git@c76f1637ce1d7a154af83495b67ead2559cd5715

Older syntax:

    git+https://github.com/JustAnotherArchivist/snscrape.git@c76f1637ce1d7a154af83495b67ead2559cd5715#egg=snscrape
